HHS/Head-Start-TTADP
# [TTAHUB-3027] Fix bad created vias and remove path from whence they sprang

## Description of change
- Remove overly complicated logic from goal form around setting createdVia. We only need set that if a goal is created there, there is no case for updating existing createdVias there.
- Fix two erroneously set createdVias. These were identified via the audit log.

## How to test
Confirm that an existing goal does not have a createdVia set when edited on the RTR, including a merged goal
Confirm that the migration on recent prod data sets those two goals to a createdVia of "merge" (their original value)

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3027


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
